truss/contexts/local_loader/truss_file_syncer.py

```diff
@@ -1,27 +1,80 @@
+import logging
 from pathlib import Path
 from threading import Thread
+from typing import Optional
 
-from truss.util.path import copy_file_path
+from truss.constants import CONFIG_FILE
+from truss.patch.calc_patch import calc_config_patches
+from truss.patch.local_truss_patch_applier import LocalTrussPatchApplier
+from truss.templates.control.control.helpers.types import (
+    Action,
+    ModelCodePatch,
+    Patch,
+    PatchType,
+)
+from truss.truss_config import TrussConfig
+from truss.truss_spec import TrussSpec
+from truss.util.path import file_content
 from watchfiles import Change, watch
 
+OP_2_ACTION = {
+    Change.added: Action.ADD,
+    Change.deleted: Action.REMOVE,
+    Change.modified: Action.UPDATE,
+}
+
+
+class TrussPatchEmitter:
+    def __init__(
+        self,
+        truss_dir: Path,
+        logger: logging.Logger,
+    ) -> None:
+        self._truss_dir = truss_dir
+        self._config = TrussConfig.from_yaml(self._truss_dir / CONFIG_FILE)
+        self._logger = logger
+
+    def __call__(self, op, path: Path) -> Optional[Patch]:
+        truss_spec = TrussSpec(self._truss_dir)
+        model_module_path = str(
+            truss_spec.model_module_dir.relative_to(self._truss_dir)
+        )
+        if str(path).startswith(model_module_path):
+            rel_path = str(path.relative_to(model_module_path))
+            return Patch(
+                type=PatchType.MODEL_CODE,
+                body=ModelCodePatch(
+                    OP_2_ACTION[op],
+                    rel_path,
+                    file_content(self._truss_dir / path),
+                ),
+            )
+        if str(path) == CONFIG_FILE:
+            new_config = TrussConfig.from_yaml(self._truss_dir / CONFIG_FILE)
+            config_patches = calc_config_patches(self._config, new_config)
```

Comment:
you probably want to up `self._config` after this line so that the next change compares to the new config